src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/hyjal_trash.cpp

```diff
@@ -1503,6 +1524,47 @@ class alliance_rifleman : public CreatureScript
 
 };
 
+class alliance_horde_guard : public CreatureScript
+{
+public:
+    alliance_horde_guard() : CreatureScript("alliance_horde_guard") { }
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return GetHyjalAI<alliance_horde_guardAI>(creature);
+    }
+
+    struct alliance_horde_guardAI : public ScriptedAI
+    {
+        alliance_horde_guardAI(Creature* creature) : ScriptedAI(creature) {}
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            Talk(TRASH_SAY_DEATH);
+            me->SetRespawnDelay(me->GetRespawnDelay()+480); // prevent respawn in combat,add wave spawn time to delay respawn
```

Comment:
>// prevent respawn in combat,add wave spawn time to delay respawn

Do they really behave like this? I mean was this confirmed?

Reply:
Generic C++ scripts for trash is something we usually remove because functions they has weren't confirmed or unnecessary

Reply:
Any suggestions to prevent they respawned between waves?

Reply:
They shouldn't?

Reply:
On Retail Server guards never respawn between waves.But in TC it does.I can't find a way to solve it.

Reply:
But all that code does is only delays respawn for a random time, it can't guarantee they will not respawn between waves

Reply:
To me it looks they don't respawn at all, meaning respawn time simply can be increased in DB to one day or 7 days
Maybe they are forced to respawn when player starts boss 2 or boss 4 but that should be done in instance script

Reply:
Do you know how to find all guids of a entry in instance?Can you give me a code example for reference?

Reply:
I recommend to first figure out how it works on retail to not implement something that may be not necessary

Reply:
Ok.I will revent this for before figure out how it works on retail vision.

Reply:
Changes have revent